daemons/based/based_callbacks.c

```diff
@@ -1522,6 +1577,18 @@ cib_process_command(xmlNode *request, const cib_operation_t *operation,
                                the_cib, *cib_diff, change_sections);
         }
 
+        /* Commit-transaction is special. We process it and activate the result
+         * only locally, but we still need to sync the updated CIB to all nodes.
+         * However, if we run the sync within cib_process_commit_transaction(),
+         * it will be rejected locally due to an older epoch in the replacement
+         * CIB. cib_perform_op() updates the epoch via xml_create_patchset()
+         * after cib_process_commit_transaction() has already returned.
+         */
+        if (pcmk__str_eq(op, PCMK__CIB_REQUEST_COMMIT_TRANSACT,
+                         pcmk__str_none)) {
+            sync_our_cib(request, TRUE);
```

Comment:
Might be good to error-check the sync, though it's not clear how practical that is -- we've already activated the new CIB by this point. We could hack around it but it might be ugly.